src/components/App.vue

```diff
@@ -0,0 +1,35 @@
+<template>
+  <div id="app">
+    <template v-if="$store.state.user">
+      <h2>You are logged in!</h2>
+      <div class="logout">
+        <button @click="logout">Logout</button>
+      </div>
+      <router-view></router-view>
```

Comment:
I believe that Vue Router is a library that you use for managing how URL paths are constructed (given Vue is a single page app)

But I don't really understand anything about how this works or what this div is doing

Reply:
Having looked at /router/index.js

The <router-view> element says: show me whatever router wants to show at this location (which is hello, because we are at /

But since this element only shows if they are logged in (thanks to the v-if statement), then non-auth won't see Hello

Does that mean the browser won't even load Hello, because v-if didn't trigger? Seems meaningful for security...

Reply:
> Does that mean the browser won't even load Hello, because v-if didn't trigger?

Let's talk about this high bandwidth.